File: rust/src/api/types/decimals.rs
//! Decimals response type for orderbook metadata.

use serde::Deserialize;

/// API response for orderbook decimal metadata.
#[derive(Debug, Clone, Deserialize)]
pub struct DecimalsResponse {
    pub orderbook_id: String,
    pub base_decimals: u8,
    pub quote_decimals: u8,
    pub price_decimals: u8,
}

File: python/src/api/types/decimals.py
"""Decimals-related types for the Lightcone REST API."""

from dataclasses import dataclass

from ..error import DeserializeError


@dataclass
class DecimalsResponse:
    """Response for GET /api/orderbook/{id}/decimals."""

    orderbook_id: str
    base_decimals: int
    quote_decimals: int
    price_decimals: int

    @classmethod
    def from_dict(cls, data: dict) -> "DecimalsResponse":
        try:
            return cls(
                orderbook_id=data["orderbook_id"],
                base_decimals=data["base_decimals"],
                quote_decimals=data["quote_decimals"],
                price_decimals=data["price_decimals"],
            )
        except KeyError as e:
            raise DeserializeError(f"Missing required field in DecimalsResponse: {e}")
